specs/simple-serialize.md

```diff
@@ -1,424 +1,116 @@
-| `little`     | Little endian.                                                                                 |
-| `byteorder`  | Specifies [endianness](https://en.wikipedia.org/wiki/Endianness): big endian or little endian. |
-| `len`        | Length/number of bytes.                                                                        |
-| `to_bytes`   | Convert to bytes. Should take parameters ``size`` and ``byteorder``.                           |
-| `from_bytes` | Convert from bytes to object. Should take ``bytes`` and ``byteorder``.                         |
-| `value`      | The value to serialize.                                                                        |
-| `rawbytes`   | Raw serialized bytes.                                                                          |
-| `deserialized_object` | The deserialized data in the data structure of your programming language.             |
-| `new_index`  | An index to keep track the latest position where the `rawbytes` have been deserialized.        |
+# SimpleSerialiZe (SSZ)
+
+This is a **work in progress** describing typing, serialization and Merkleization of Ethereum 2.0 objects.
+
+## Table of contents
+
+- [Constants](#constants)
+- [Typing](#typing)
+    - [Basic types](#basic-types)
+    - [Composite types](#composite-types)
+    - [Aliases](#aliases)
+- [Serialization](#serialization)
+    - [`uintN`](#uintn)
+    - [`bool`](#bool)
+    - [Containers, tuples, lists](#containers-tuples-lists)
+- [Deserialization](#deserialization)
+- [Merkleization](#merkleization)
+- [Self-signed containers](#self-signed-containers)
+- [Implementations](#implementations)
 
 ## Constants
 
-| Constant          | Value | Definition                                                                            |
-|:------------------|:-----:|:--------------------------------------------------------------------------------------|
-| `LENGTH_BYTES`    |   4   | Number of bytes used for the length added before a variable-length serialized object. |
-| `SSZ_CHUNK_SIZE`  |  128  | Number of bytes for the chunk size of the Merkle tree leaf.                           |
+| Name | Value | Description |
+|-|-|-|
+| `BYTES_PER_CHUNK` | `32` | Number of bytes per chunk.
+| `BYTES_PER_LENGTH_PREFIX` | `4` | Number of bytes per serialized length prefix. |
 
-## Overview
+## Typing
+### Basic types
 
-### Serialize/Encode
+* `uintN`: `N`-bit unsigned integer (where `N in [8, 16, 32, 64, 128, 256]`)
+* `bool`: `True` or `False`
 
-#### uintN
+### Composite types
 
-| uint Type | Usage                                                      |
-|:---------:|:-----------------------------------------------------------|
-| `uintN`   | Type of `N` bits unsigned integer, where ``N % 8 == 0``.   |
+* **container**: ordered heterogenous collection of values
+    * key-pair curly bracket notation `{}`, e.g. `{'foo': "uint64", 'bar': "bool"}`
+* **tuple**: ordered fixed-length homogeneous collection of values
```

Comment:
aren't these better called `array`:s? `tuples` tend to allow heterogeneous types

Reply:
I don't care either way. Implementations can use whatever terminology works best. For example, the natural terminology for Go is "array" (instead of tuple) and "slice" (instead of list). Notice that in Python lists also allow heterogeneous types 🤷‍♂️